packages/mediapipe-core/lib/src/task_options.dart

```diff
@@ -0,0 +1,125 @@
+import 'dart:ffi';
+import 'dart:typed_data';
+import 'package:equatable/equatable.dart';
+import 'package:ffi/ffi.dart';
+import 'mediapipe_common_bindings.dart' as bindings;
+import 'ffi_utils.dart';
+
+class BaseOptions extends Equatable {
+  const BaseOptions({this.modelAssetBuffer, this.modelAssetPath})
+      : assert(
+          !(modelAssetBuffer == null && modelAssetPath == null),
+          'You must supply either `modelAssetBuffer` or `modelAssetPath`',
+        ),
+        assert(
+          !(modelAssetBuffer != null && modelAssetPath != null),
+          'You must only supply one of `modelAssetBuffer` and `modelAssetPath`',
+        );
+
+  /// The model asset file contents as bytes;
+  final Uint8List? modelAssetBuffer;
+
+  /// Path to the model asset file.
+  final String? modelAssetPath;
+
+  Pointer<bindings.BaseOptions> toStruct() {
+    final struct = calloc<bindings.BaseOptions>();
+
+    if (modelAssetPath != null) {
+      struct.ref.model_asset_path = prepareString(modelAssetPath!);
+    }
+    if (modelAssetBuffer != null) {
+      struct.ref.model_asset_buffer = prepareUint8List(modelAssetBuffer!);
+    }
+    return struct;
+  }
+
+  @override
+  List<Object?> get props => [modelAssetBuffer, modelAssetPath];
+}
+
+class ClassifierOptions extends Equatable {
+  const ClassifierOptions({
+    this.displayNamesLocale,
+    this.maxResults,
+    this.scoreThreshold,
+    this.categoryAllowlist,
+    this.categoryDenylist,
+  });
+
+  /// The locale to use for display names specified through the TFLite Model
+  /// Metadata.
+  final String? displayNamesLocale;
+
+  /// The maximum number of top-scored classification results to return.
+  final int? maxResults;
+
+  /// Overrides the ones provided in the model metadata.
+  ///
+  /// Results below this value are rejected.
+  final double? scoreThreshold;
+
+  /// Allowlist of category names.
```

Comment:
Do these correlate to [Category.categoryName]s? Cross references would be great with some breadcrumbing if so